File: 215-kth-largest-element-in-an-array/solution.py
import heapq


class Solution:
  def findKthLargest(self, nums: list[int], k: int) -> int:
    largest: list[int] = []
    for num in nums:
      if len(largest) < k:
        heapq.heappush(largest, num)
        continue

      if num > largest[0]:
        heapq.heappop(largest)
        heapq.heappush(largest, num)

    return largest[0]

File: 215-kth-largest-element-in-an-array/solution.cpp
#include <vector>
#include <queue>
#include <functional>

class Solution
{
public:
  int findKthLargest(std::vector<int> &nums, int k)
  {
    std::priority_queue<int, std::vector<int>, std::greater<int>> largest;
    for (const auto num : nums)
    {
      if (largest.size() < k)
      {
        largest.emplace(num);
        continue;
      }

      if (num > largest.top())
      {
        largest.pop();
        largest.emplace(num);
      }
    }

    return largest.top();
  }
};